src/basic/errno-util.h

```diff
@@ -16,6 +16,11 @@
  * Note that we use the GNU variant of strerror_r() here. */
 #define STRERROR(errnum) strerror_r(abs(errnum), (char[ERRNO_BUF_LEN]){}, ERRNO_BUF_LEN)
 
+/* A helper to print an error message or message for functions that return 0 on EOF.
+ * Note that we can't use ({ … }) to define a temporary variable, so errnum is
+ * evaluated twice. */
+#define STRERROR_OR_EOF(errnum, message) (errnum) != 0 ? STRERROR(errnum) : (message)
```

Comment:
should really be enclosed in (), because otherwise his will work:

```c
printf("%s\n", 5 + STRERROR_OR_EOF(EINTR, "foobar");
```

and i am pretty sure it really shouldn't work

Reply:
honestly i would hardcode the string to "Unexpected EOF". That is an appropriate string for all users of this helper, even if the current strings we use for them are a bit different.

Reply:
Done.